pkg/rlptools/overrides.go

```diff
@@ -37,7 +39,8 @@ func ApplyOverrides(topology *kuadrantgatewayapi.Topology, gateway *gatewayapiv1
 
 	for _, route := range topology.Routes() {
 		if !slices.Contains(kuadrantgatewayapi.GetRouteAcceptedGatewayParentKeys(route.HTTPRoute), client.ObjectKeyFromObject(gateway)) {
-			overriddenPolicies = append(overriddenPolicies, route.AttachedPolicies()...)
+			routePolicies := utils.Map(route.AttachedPolicies(), func(pNode kuadrantgatewayapi.PolicyNode) kuadrantgatewayapi.Policy { return pNode.Policy })
```

Comment:
TO avoid adding another dependency:

```suggestion
			routePolicies := lo.Map(route.AttachedPolicies(), func(pNode kuadrantgatewayapi.PolicyNode, _ int) kuadrantgatewayapi.Policy { return pNode.Policy })
```

Reply:
same as https://github.com/Kuadrant/kuadrant-operator/pull/751#discussion_r1672144674

Reply:
Maybe, for other PR, we could remove all `utils.Map` and `utils.Filter` and rewrite using [github.com/samber/lo](https://pkg.go.dev/github.com/samber/lo) (or something else). Then we can remove `utils` code and have the external dependency only.